### PR TITLE
Updating certificate recipe to copy compute certs to compute

### PR DESCRIPTION
**Pull request by msumankumar**

### CODE REVIEW

File: components/cookbooks/certificate/recipes/add.rb [hunk 1]
```diff
@@ -12,10 +12,27 @@
...
-	expires_on = node[:expiry_time]
+     run_context.include_recipe provider + "::add_certificate"
+     expires_on = node[:expiry_time]
+     if node.workorder.payLoad.has_key?("DependsOn")
+           cer_cont = node[:cert_content]
+           key_cont = node[:key_content]
+           ca_cont = node[:ca_content]
+           pfx_cont = node[:pfx_content]
+           `mkdir -p #{path}`
+           File.open("#{path}/cert_content.pem", "w") {|line| line.puts "\r" + cer_cont.to_s }
```
Comment (sureshg) [inline components/cookbooks/certificate/recipes/add.rb]: IMO, no need for `content` in the file name. Usually, it's using the format
* `common-name.crt` or `.pem`
* `common-name.key`
* `ca_cert.crt` 
* `common-name.pfx`

File: components/cookbooks/certificate/recipes/add.rb [hunk 2]
```diff
@@ -12,10 +12,27 @@
 end
 
 expires_on = node[:expiry_time]
-
+cer_cont = ""
+key_cont = ""
+ca_cont = ""
+pfx_cont = ""
```
Comment (sureshg) [inline components/cookbooks/certificate/recipes/add.rb]: No need to init these vars. Remove them.

File: components/cookbooks/certificate/recipes/add.rb [hunk 3]
```diff
@@ -12,10 +12,24 @@
 end
 
 expires_on = node[:expiry_time]
-
+cert_path = node[:certificate][:path]
+ci_name = node[:app_name]
+path = "#{cert_path}/#{ci_name}"
+cname = node.workorder.rfcCi.ciAttributes.common_name
 if !auto_provision.nil? && auto_provision == "true" && !provider.nil? && !provider.empty?
```
Comment (sureshg) [inline components/cookbooks/certificate/recipes/add.rb]: Shouldn't we add the check for managed_via to make sure it's created on remote compute?

File: components/cookbooks/certificate/recipes/add.rb [hunk 4]
```diff
@@ -12,10 +12,24 @@
...
-	include_recipe provider + "::add_certificate" 
-	expires_on = node[:expiry_time]
+     run_context.include_recipe provider + "::add_certificate"
+     expires_on = node[:expiry_time]
+       if node.workorder.payLoad.has_key?("DependsOn")
```
Comment (msumankumar) [inline components/cookbooks/certificate/recipes/add.rb]: @sureshg Used this condition `if defined?(node.workorder.payLoad.ManagedVia)` for some reason this was not working so used "DependsOn"

**Comment by jayeshpise** [comment]
@msumankumar : Lets make the persist cert logic conditional based on value node[:persist_cert]
Also, can you modify the secrets-client recipe accordingly to set that value to false so that the certs don't get saved on the compute in that case

**Comment by gauravlall** [comment]
@msumankumar  We need kitchen ci test

**Comment by msumankumar** [comment]
Added the kitchen CI test cases

**Pull request closed without merging**